src/main/java/com/umbr3114/controllers/DropController.java

```diff
@@ -328,4 +328,45 @@ public String provideOwnerId() {
 
         return dropViewModel;
     });
+
+    /**
+     * to search for drops or posts
+     */
+    public static Route search_drops_posts = ((request, response) -> {
+
+        String userInput;
+        SearchModel searchResults = new SearchModel();
+        RequestParamHelper params = new RequestParamHelper(request);
+        userInput = params.valueOf("userInput");
+
+        FindIterable<DropModel> iterable;
+        FindIterable<PostModel> iterable2;
+
+        if(userInput ==  null){
+            halt(HttpStatus.FORBIDDEN_403, new GeneralResponse(HttpStatus.FORBIDDEN_403,
+                    "no input").toJSON());
+        }
+        JacksonMongoCollection<DropModel> dropCollection = new CollectionFactory<DropModel>(Main.services.dbService(),
+                DropModel.class).getCollection();
+
+        dropCollection.createIndex(Indexes.text("title"));
+        iterable = dropCollection.find(Filters.text(userInput));
+        if(iterable ==  null){
```

Comment:
this case will probably never happen, what will happen is that an empty iterable is going to be returned.